llvm/lib/Target/STM8/STM8CallingConv.td

```diff
@@ -0,0 +1,32 @@
+//==- STM8CallingConv.td - Calling Conventions for STM8 -*- tablegen -*-==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// This describes the calling conventions for STM8 architecture.
+//===----------------------------------------------------------------------===//
+
+//===----------------------------------------------------------------------===//
+// STM8 Return Value Calling convention
+//===----------------------------------------------------------------------===//
+def RetCC_STM8 : CallingConv<[
```

Comment:
We'll problably want to support other calling conventions at some point, so I think we should give it a name that indicates what particular calling convention it is. I'd suggest `RetCC_STM8_SDCCv1` and `CC_STM8_SDCCv1_VarArgs`.